server/src/com/thoughtworks/go/config/CachedFileGoConfig.java

```diff
@@ -95,9 +96,12 @@ public void onTimer() {
 
     private synchronized void loadFromDisk() {
```

Comment:
What I am worried about are more deadlocks. The synchronized method actually takes a lock (or monitor) on the object, and the synchronized block you have takes a lock on the other object (output of "mutex"). So, if some method is synchronized and another is not, in the same class, then it can deadlock. I didn't find any such usages today, but it might be worth (later?) looking at whether they need to be synchronized. There are two locks in this class, when only one should exist, I feel.